Local navigation: Add component (#1468)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/content/docs/5.3/components/local-navigation.md

---
layout: docs
title: Local navigation
description: Use local navigation to add a navigation that will nicely wrap on small viewports.
group: components
aliases:
  - "/docs/components/local-navigation/"
toc: true
added: "5.3"
---

## Example

Resize your browser to see the behavior of this component.

{{< callout info >}}

> Since the component is supposed to be rendered on a full size page and not inside an example with padding, there might be some rendering issues in mobile viewport.

Reviewer: ```suggestion
Since the component is supposed to be rendered on a full-size page, there are some slight rendering issues in the mobile viewport in the context of this documentation page.
```

    <span class="container-xxl">Shop</span>
  </button>
  <div id="collapseLocalNavDark" class="container-xxl local-nav-collapse d-lg-block collapse">
      <div class="navbar navbar-dark navbar-expand-lg">
        <ul class="navbar-nav w-100">
          <li class="nav-item"><a class="nav-link" href="#"><div class="container-xxl">Discover</div></a></li>
          <li class="nav-item"><a class="nav-link active" href="#" aria-current="page"><div class="container-xxl">Shop</div></a></li>
          <li class="nav-item"><a class="nav-link" href="#"><div class="container-xxl">Services</div></a></li>
        </ul>
      </div>
  </div>
</nav>
{{< /example >}}

## CSS

### Variables

Values for the CSS variables are set via Sass, so Sass customization is still supported, too.

{{< scss-docs name="local-nav-css-vars" file="scss/_local-navigation.scss" >}}

Customization through CSS variables can be seen on the `.local-nav-dark` class where we override specific values without adding duplicate CSS selectors.

{{< scss-docs name="local-nav-dark-css-vars" file="scss/_local-navigation.scss" >}}

### Sass variables

> Variables for all local navigation:

Reviewer: ```suggestion
Variables for all local navigations:
```

Author: It introduces a cspell issue, should it still be merged ?

Reviewer: > The noun navigation can be countable or uncountable.
>
> In more general, commonly used, contexts, the plural form will also be navigation.
>
> However, in more specific contexts, the plural form can also be navigations e.g. in reference to various types of navigations or a collection of navigations.

I suppose it can be considered as a collection of variants, so with an "s".

---
layout: docs
title: Local navigation
description: Use local navigation to add a navigation that will nicely wrap on small viewports.
group: components
aliases:
  - "/docs/components/local-navigation/"
toc: true
added: "5.3"
---

## Example

Resize your browser to see the behavior of this component.

{{< callout info >}}
Since the component is supposed to be rendered on a full-size page, there are some slight rendering issues in the mobile viewport in the context of this documentation page.
{{< /callout >}}

{{< example >}}
<nav class="local-nav" aria-label="Basic local navigation">
  <button class="local-nav-button collapsed d-lg-none" type="button" data-bs-toggle="collapse" data-bs-target="#collapseLocalNav" aria-expanded="false" aria-controls="collapseLocalNav">
    <span class="container-xxl">Shop</span>
  </button>
  <div id="collapseLocalNav" class="container-xxl local-nav-collapse d-lg-block collapse">
      <div class="navbar navbar-expand-lg">
        <ul class="navbar-nav w-100">

>           <li class="nav-item"><a class="nav-link" href="#"><div class="container-xxl">Discover</div></a></li>

Reviewer: In this collapse navbar, shouldn't we use `<span class="container-xxl">` instead of `<div class="container-xxl">` everywhere? It's used like this within the `<button>` that triggers the collapse, and it seems to be more logical in the tree to have spans within an `<li>` > `<a>` rather than a `<div>`.
